commonmark-ext-styles/src/main/java/org/commonmark/ext/styles/internal/StylesDelimiterProcessor.java

```diff
@@ -0,0 +1,51 @@
+package org.commonmark.ext.styles.internal;
+
+import org.commonmark.ext.styles.Styles;
+import org.commonmark.node.Node;
+import org.commonmark.node.Text;
+import org.commonmark.parser.delimiter.DelimiterProcessor;
+import org.commonmark.parser.delimiter.DelimiterRun;
+
+public class StylesDelimiterProcessor implements DelimiterProcessor {
```

Comment:
Implementing it as a delimiter processor + attribute provider is clever :)! I think if we supported extending inline parsing (see #113), this would probably a good use for it. But we don't, so a delimiter processor seems nice.

The only problem I see is that it means `{` can't be used by other delimiter processors anymore. I would probably mention that in the README.

Oh, one more thing, does having spaces inside the braces work?: `![](file.jpg){ width=50 }`

Reply:
Added note to the README.

Reply:
> Oh, one more thing, does having spaces inside the braces work?: `![](file.jpg){ width=50 }`

Unfortunately no, as it seems that `InlineParserImpl` will not recognise the opening delimiter if it is followed by whitespace:
https://github.com/atlassian/commonmark-java/blob/cb4bee61046e21a0bef4ab96860334d1f089fba7/commonmark/src/main/java/org/commonmark/internal/InlineParserImpl.java#L703,L715